Add output_size argument to caffe2 Int8ResizeNearest

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30202 Add output_size argument to caffe2 Int8ResizeNearest**

Summary:
Pytorch Upsample operator has output_size as an argument.
For quantized tensor inputs we cannot get the input_size to calculate the width and height scale factor.
Instead we pass the output_size directly to caffe2 to calculate the scale factors.

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_upsample

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D18631478](https://our.internmc.facebook.com/intern/diff/D18631478)

supriyar committed:
Add output_size argument to caffe2 Int8ResizeNearest
Summary:
Pytorch Upsample operator has output_size as an argument.
For quantized tensor inputs we cannot get the input_size to calculate the width and height scale factor.
Instead we pass the output_size directly to caffe2 to calculate the scale factors.

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_upsample

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "Add output_size argument to caffe2 Int8ResizeNearest"


Summary:
Pytorch Upsample operator has output_size as an argument.
For quantized tensor inputs we cannot get the input_size to calculate the width and height scale factor.
Instead we pass the output_size directly to caffe2 to calculate the scale factors.

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_upsample

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "Add output_size argument to caffe2 Int8ResizeNearest"


Summary:
Pytorch Upsample operator has output_size as an argument.
For quantized tensor inputs we cannot get the input_size to calculate the width and height scale factor.
Instead we pass the output_size directly to caffe2 to calculate the scale factors.

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_upsample

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "Add output_size argument to caffe2 Int8ResizeNearest"


Summary:
Pytorch Upsample operator has output_size as an argument.
For quantized tensor inputs we cannot get the input_size to calculate the width and height scale factor.
Instead we pass the output_size directly to caffe2 to calculate the scale factors.

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_upsample

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D18631478](https://our.internmc.facebook.com/intern/diff/D18631478)

[ghstack-poisoned]

houseroad commented on caffe2/operators/quantized/int8_resize_nearest_op.cc:
```diff
@@ -12,6 +12,7 @@ OPERATOR_SCHEMA(Int8ResizeNearest)
     .Arg("Y_zero_point", "Output tensor quantization offset")
     .Arg("width_scale", "Scale along width dimension")
     .Arg("height_scale", "Scale along height dimension")
+    .Arg("output_size", "Output dimensions (HxW)")
```
Let's mention when output_size and scale both exist, which one will be the source of the truth.

houseroad commented on caffe2/operators/quantized/int8_resize_nearest_op.h:
```diff
@@ -17,6 +17,7 @@ class Int8ResizeNearestOp final : public Operator<CPUContext> {
       : Operator<CPUContext>(std::forward<Args>(args)...) {
     width_scale_ = this->template GetSingleArgument<float>("width_scale", 1);
     height_scale_ = this->template GetSingleArgument<float>("height_scale", 1);
+    output_dims = this->template GetRepeatedArgument<int>("output_size");
```
let's give it a default value, if this is missing, we probably should not crash.

houseroad reviewed (approved):
LGTM, some inline comments. Please rebase to the master, so broken builds can pass.

supriyar committed:
Update on "Add output_size argument to caffe2 Int8ResizeNearest"


Summary:
Pytorch Upsample operator has output_size as an argument.
For quantized tensor inputs we cannot get the input_size to calculate the width and height scale factor.
Instead we pass the output_size directly to caffe2 to calculate the scale factors.

Test Plan:
python test/onnx/test_pytorch_onnx_caffe2_quantized.py TestQuantizedOps.test_upsample

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D18631478](https://our.internmc.facebook.com/intern/diff/D18631478)

[ghstack-poisoned]
